frontend/src/metabase/query_builder/components/QueryModals.tsx

```diff
@@ -0,0 +1,356 @@
+import React from "react";
+import { connect } from "react-redux";
+
+import { t } from "ttag";
+import _ from "underscore";
+
+import Questions from "metabase/entities/questions";
+import { ROOT_COLLECTION } from "metabase/entities/collections";
+
+import { MODAL_TYPES } from "metabase/query_builder/constants";
+
+import Modal from "metabase/components/Modal";
+
+import SaveQuestionModal from "metabase/containers/SaveQuestionModal";
+import QuestionSavedModal from "metabase/components/QuestionSavedModal";
+import AddToDashSelectDashModal from "metabase/containers/AddToDashSelectDashModal";
+
+import CollectionMoveModal from "metabase/containers/CollectionMoveModal";
+import ArchiveQuestionModal from "metabase/questions/containers/ArchiveQuestionModal";
+import QuestionEmbedWidget from "metabase/query_builder/containers/QuestionEmbedWidget";
+
+import { CreateAlertModalContent } from "metabase/query_builder/components/AlertModals";
+import { ImpossibleToCreateModelModal } from "metabase/query_builder/components/ImpossibleToCreateModelModal";
+import NewDatasetModal from "metabase/query_builder/components/NewDatasetModal";
+import EntityCopyModal from "metabase/entities/containers/EntityCopyModal";
+import BulkFilterModal from "metabase/query_builder/components/filters/modals/BulkFilterModal";
+import NewEventModal from "metabase/timelines/questions/containers/NewEventModal";
+import EditEventModal from "metabase/timelines/questions/containers/EditEventModal";
+import MoveEventModal from "metabase/timelines/questions/containers/MoveEventModal";
+import PreviewQueryModal from "metabase/query_builder/components/view/PreviewQueryModal";
+import ConvertQueryModal from "metabase/query_builder/components/view/ConvertQueryModal";
+import QuestionMoveToast from "metabase/questions/components/QuestionMoveToast";
+import { Card, Collection, User } from "metabase-types/api";
+import StructuredQuery from "metabase-lib/queries/StructuredQuery";
+import Question from "metabase-lib/Question";
+
+const mapDispatchToProps = {
+  setQuestionCollection: Questions.actions.setCollection,
+};
+
+type ModalType = typeof MODAL_TYPES[keyof typeof MODAL_TYPES];
+
+interface QueryModalsProps {
+  questionAlerts: any;
+  user: User;
+  modal: ModalType;
+  modalContext: number;
+  question: Question;
+  initialCollectionId: number;
+  updateQuestion: (question: Question, config?: object) => void;
+  setQueryBuilderMode: (mode: string) => void;
+  originalQuestion: Question;
+  card: Card;
+  onCreate: (question: Question, flag?: boolean) => void;
+  onSave: (question: Question, flag?: boolean) => void;
```

Comment:
Not sure if these types match the current state of the query builder:

1. [`onCreate` accepts a single argument](https://github.com/metabase/metabase/blob/7b034eb869d9aa0958e228f423c0627415dfb58c/frontend/src/metabase/query_builder/containers/QueryBuilder.jsx#L267)
2. [`onSave` accepts two arguments, but the second one should be an object](https://github.com/metabase/metabase/blob/7b034eb869d9aa0958e228f423c0627415dfb58c/frontend/src/metabase/query_builder/containers/QueryBuilder.jsx#L277)

I could be wrong here, though — QB prop drilling is 🫠

Reply:
maybe you're right here, I don't fully understand prop drilling as well, but from what I see, we do not touch directly lines you provided here, I tried to record explanation here https://www.loom.com/share/82b44c7bf89a4fcb8c6ab8eb7fb0216f

second half of the video can be skipped 😄

Reply:
I suppose these signatures are just outdated; I assume the boolean argument was used to trigger query re-runs. And now, `QueryModals` either passes `false` as a second arg for `onCreate` and `onSave`, or doesn't pass the flag at all. Given that `onCreate` only accepts a single argument and `onSave` expects an optional object, both these signatures have no effect.

Also, doesn't seem like there's any intermediate component that could wrap `onCreate` and `onSave` into anything that accepts a boolean flag. `QueryModals` is only rendered by QB's `View` component ([here](https://github.com/metabase/metabase/blob/7b034eb869d9aa0958e228f423c0627415dfb58c/frontend/src/metabase/query_builder/components/view/View.jsx#L514) and [here](https://github.com/metabase/metabase/blob/7b034eb869d9aa0958e228f423c0627415dfb58c/frontend/src/metabase/query_builder/components/view/View.jsx#L468)). In both cases, `View` just spreads all its props for the modals component. And `View` is a [direct child of the `QueryBuilder` component](https://github.com/metabase/metabase/blob/7b034eb869d9aa0958e228f423c0627415dfb58c/frontend/src/metabase/query_builder/containers/QueryBuilder.jsx#L402) that defines both `onCreate` and `onSave`

Reply:
that's a great knowledge sharing 😄 thanks for so deep digging into code